test/server/models/TransactionSettlement.test.ts

```diff
@@ -0,0 +1,201 @@
+import { expect } from 'chai';
+import { times } from 'lodash';
+
+import { TransactionKind } from '../../../server/constants/transaction-kind';
+import { FEES_ON_TOP_TRANSACTION_PROPERTIES } from '../../../server/constants/transactions';
+import models, { Op } from '../../../server/models';
+import TransactionSettlement, { TransactionSettlementStatus } from '../../../server/models/TransactionSettlement';
+import {
+  fakeCollective,
+  fakeExpense,
+  fakeHost,
+  fakeOrganization,
+  fakeTransaction,
+  fakeUser,
+  fakeUUID,
+} from '../../test-helpers/fake-data';
+import * as utils from '../../utils';
+
+// ---- Helpers ----
+
+const fakeContribution = (fromCollective, collective, transactionGroup) => {
+  return fakeTransaction(
+    {
+      TransactionGroup: fakeUUID(transactionGroup),
+      kind: TransactionKind.CONTRIBUTION,
+      CollectiveId: collective.id,
+      FromCollectiveId: fromCollective.id,
+      amount: 1000,
+      type: 'CREDIT',
+      description: 'A simple contribution **without** platform tip',
+    },
+    {
+      createDoubleEntry: true,
+    },
+  );
+};
+
+// Insert contributions **with platform tips**, using the new format described in https://github.com/opencollective/opencollective/issues/4124.
+// As writing this, the rest of the code is not yet using this new format.
+const fakeContributionWithPlatformTipNewFormat = async (fromCollective, collective, transactionGroup) => {
+  // Base transaction
+  const transaction = await fakeTransaction(
+    {
+      TransactionGroup: fakeUUID(transactionGroup),
+      kind: TransactionKind.CONTRIBUTION,
+      FromCollectiveId: fromCollective.id,
+      CollectiveId: collective.id,
+      HostCollectiveId: collective.isHostAccount ? collective.id : collective.HostCollectiveId,
+      amount: 1000,
+      description: 'A contribution **with** platform tip',
+    },
+    {
+      createDoubleEntry: true,
+    },
+  );
+
+  // DEBIT Tip transaction (contributor -> OC)
+  await fakeTransaction({
+    TransactionGroup: transaction.TransactionGroup,
+    kind: TransactionKind.PLATFORM_TIP,
+    type: 'DEBIT',
+    amount: -200,
+    CollectiveId: transaction.FromCollectiveId,
+    FromCollectiveId: FEES_ON_TOP_TRANSACTION_PROPERTIES.CollectiveId,
+    HostCollectiveId: null,
+    description: 'Tip transaction form the contributor to Open Collective',
```

Comment:
Replace form by from here and everywhere else?

Reply:
Good catch!